ipykernel/ipkernel.py

```diff
@@ -54,8 +54,7 @@ def __init__(self, **kwargs):
         # TMP - hack while developing
         self.shell._reply_content = None
 
-        self.comm_manager = CommManager(shell=self.shell, parent=self,
-                                        kernel=self)
+        self.comm_manager = CommManager(kernel=self)
```

Comment:
Ok, what's the reason for needing parent set?

If it's required, we should just remove the `kernel` arg and use parent internally then.